integration/e2e_test.go

```diff
@@ -299,3 +302,89 @@ func deployMicrobot(t testing.TB) {
 		}
 	}
 }
+
+func upgradeCluster(t testing.TB) {
+	version := os.Getenv("CAPI_UPGRADE_VERSION")
+	if version == "" {
+		t.Fatalf("Environment variable CAPI_UPGRADE_VERSION is not set." +
+			"Please set it to the version you want to upgrade to.")
+	}
+
+	control_plane_name := os.Getenv("CAPI_UPGRADE_CP_NAME")
+	if control_plane_name == "" {
+		t.Fatalf("Environment variable CAPI_UPGRADE_CP_NAME is not set." +
+			"Please set it to the name of the control plane you want to upgrade.")
+	}
+
+	control_plane_type := os.Getenv("CAPI_UPGRADE_CP_TYPE")
+	if control_plane_type == "" {
+		t.Fatalf("Environment variable CAPI_UPGRADE_CP_TYPE is not set." +
+			"Please set it to the type of the control plane you want to upgrade.")
+	}
+
+	worker_deployment_name := os.Getenv("CAPI_UPGRADE_MD_NAME")
+	if worker_deployment_name == "" {
+		t.Fatalf("Environment variable CAPI_UPGRADE_MD_NAME is not set." +
+			"Please set it to the name of the machine deployment you want to upgrade.")
+	}
+
+	worker_deployment_type := os.Getenv("CAPI_UPGRADE_MD_TYPE")
+	if worker_deployment_type == "" {
+		t.Fatalf("Environment variable CAPI_UPGRADE_MD_TYPE is not set." +
+			"Please set it to the type of the machine deployment you want to upgrade.")
+	}
+
+	t.Logf("Upgrading cluster to %s", version)
+	// Patch contol plane machine upgrades.
+	patch := fmt.Sprintf(`{"spec":{"version":"%s"}}`, version)
+	command := []string{"kubectl", "patch", "--type=merge", control_plane_type, control_plane_name, "--patch", patch}
```

Comment:
no need for a separate variable 
```suggestion
	command := []string{"kubectl", "patch", "--type=merge", control_plane_type, control_plane_name, "--patch", fmt.Sprintf(`{"spec":{"version":"%s"}}`, version)
}
```